### PR TITLE
Add documentation for monster room datamap

### PR DESCRIPTION
**Pull request by GingerIndustries**
See neoforged/neoforge#918.

------------------
Preview URL: https://pr-66.neoforged-docs-previews.pages.dev

### CODE REVIEW

File: docs/datamaps/neo_maps.md [hunk 1]
```diff
@@ -55,6 +55,28 @@ Vanilla adds a burn time to logs and planks in the `minecraft:logs` and `minecra
 However, the removal does not affect any values added by other packs or mods, so if you want to change the values for the wood tags you will need to add a removal for the non flammable tag yourself.
 :::
 
+## `neoforge:monster_room_mobs`
```
Comment (ChampionAsh5357) [inline docs/datamaps/neo_maps.md:58]: New line after header.

File: docs/datamaps/neo_maps.md [hunk 2]
```diff
@@ -55,6 +55,28 @@ Vanilla adds a burn time to logs and planks in the `minecraft:logs` and `minecra
...
+NeoForge provides a data map that allows configuring the mobs which may appear in the mob spawner in a monster room, as a replacement for `MonsterRoomFeature#MOBS` (which is now ignored).
+This data map is located at `neoforge/data_maps/entity_type/monster_room_mobs.json` and its objects have the following structure:
```
Comment (ChampionAsh5357) [inline docs/datamaps/neo_maps.md:60]: Newline between codeblocks and text.

File: docs/datamaps/neo_maps.md [hunk 3]
```diff
@@ -55,6 +55,28 @@ Vanilla adds a burn time to logs and planks in the `minecraft:logs` and `minecra
...
+```js
```
Comment (ChampionAsh5357) [inline docs/datamaps/neo_maps.md:61]: Use json5

File: docs/datamaps/neo_maps.md [hunk 4]
```diff
@@ -55,6 +55,28 @@ Vanilla adds a burn time to logs and planks in the `minecraft:logs` and `minecra
...
+{
+    // The weight of this mob, relative to other mobs in the datamap
+    "weight": 100
+}
+```
+
+Example:
```
Comment (ChampionAsh5357) [inline docs/datamaps/neo_maps.md:68]: Newline between codeblocks and text.

File: docs/datamaps/neo_maps.md [hunk 5]
```diff
@@ -55,6 +55,28 @@ Vanilla adds a burn time to logs and planks in the `minecraft:logs` and `minecra
...
+```js
```
Comment (ChampionAsh5357) [inline docs/datamaps/neo_maps.md:69]: Use json5

File: docs/datamaps/neo_maps.md (hunk 2 above)
Comment (ChampionAsh5357) [inline docs/datamaps/neo_maps.md:60]: Paragraphs should be a single line, not spread across multiple.

**Review by ChampionAsh5357 — CHANGES_REQUESTED** [review]
Because I don't want to deal with more formatting changes, please address them.

**Review by ChampionAsh5357 — APPROVED** [review]
Looks good to me based on the other PR. If there are no other complaints, will merge by the end of the day.

**Review by Matyrobbrt — APPROVED** [review]

**Pull request merged**